### PR TITLE
chore: add discord example

### PR DESCRIPTION
**Pull request by Q1w1N**
### Summary
new example for discord integration

### Test plan
tested with discord

### CODE REVIEW

File: .changeset/cuddly-mangos-train.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@callstack/byorg-discord': minor
+'discord-example': patch
```
Comment (mdjastrzebski) [inline .changeset/cuddly-mangos-train.md:3]: [nit] implementing example actually accounts as a minor ;-)
```suggestion
'discord-example': minor
```

File: examples/discord/package.json [hunk 2]
```diff
@@ -15,13 +15,14 @@
   "dependencies": {
     "@ai-sdk/openai": "^1.0.2",
     "@callstack/byorg-core": "0.4.2",
```
Comment (mdjastrzebski) [inline examples/discord/package.json:17]: can you replace all 0.4.2 references with workspace? There was a bug if you released it with changeset, but now we do release it with `pnpm` so we can use workspace

File: packages/discord/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@callstack/byorg-discord",
-  "version": "0.4.2",
+  "version": "0.4.3",
```
Comment (mdjastrzebski) [inline packages/discord/package.json:3]: Leave version updating to changset

File: packages/discord/src/index.ts [hunk 4]
```diff
@@ -6,12 +6,18 @@ const UPDATE_THROTTLE_TIME = 200;
...
+  botToken: string;
 };
 
-export async function createDiscordApp(options: DiscordApplicationConfig) {
-  const { app } = options;
+type ClientStart = {
+  start: () => Promise<string>;
+};
+
+export type DiscordClient = Client & ClientStart;
```
Comment (mdjastrzebski) [inline packages/discord/src/index.ts:16]: Add code comment documenting this hack

File: packages/discord/src/index.ts [hunk 5]
```diff
@@ -76,7 +82,9 @@ export async function createDiscordApp(options: DiscordApplicationConfig) {
     await pendingEffects;
   });
 
-  await client.login(botToken);
+  client.start = () => {
```
Comment (mdjastrzebski) [inline packages/discord/src/index.ts:85]: Why are we adding methods to existing discord object?

Comment (mdjastrzebski) [inline packages/discord/src/index.ts:85]: I do not see any code in this PR actually calling this method. Is it needed?

**Review by mdjastrzebski — CHANGES_REQUESTED** [review]
I have some questions in the comments

File: packages/discord/src/index.ts (hunk 5 above)
Comment (Q1w1N) [reply packages/discord/src/index.ts:85]: No added

File: .changeset/cuddly-mangos-train.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+'@callstack/byorg-discord': minor
+'discord-example': minor
+---
+
+changed passing bot token variable and discord login
```
Comment (mdjastrzebski) [inline .changeset/cuddly-mangos-train.md:6]:
```suggestion
discord: changed passing bot token variable and discord login
```

**Review by mdjastrzebski — APPROVED** [review]

**Review by mdjastrzebski — APPROVED** [review]
Approved!

**Pull request merged**